### PR TITLE
Testz and testnzc allow with butsign constant allow better impleementattions of ant and none in avx and sse4.1

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/boost/simd/arch/x86/avx/simd/function/any.hpp [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+
+  BOOST_DISPATCH_OVERLOAD ( any_
+                          , (typename A0)
+                          , bs::avx_
+                          , bs::pack_<bd::integer_<A0>, bs::avx_>
+                         )
+  {
+    BOOST_FORCEINLINE bool operator() ( const A0 & a0) const BOOST_NOEXCEPT
+    {
+      return (_mm256_testnzc_si256(a0, Allbits<A0>()));
```
Comment (jfalcou) [inline include/boost/simd/arch/x86/avx/simd/function/any.hpp]: those parens are unrequited, did you wanted to write a cast to bool ?

File: include/boost/simd/arch/x86/avx/simd/function/any.hpp [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+  BOOST_DISPATCH_OVERLOAD ( any_
+                          , (typename A0)
+                          , bs::avx_
+                          , bs::pack_<bd::floating_<A0>, bs::avx_>
+                         )
+  {
+    BOOST_FORCEINLINE bool operator() ( const A0 & a0) const BOOST_NOEXCEPT
+    {
+      using iA0 = bd::as_integer_t<A0>;
+      return (_mm256_testnzc_si256(bitwise_cast<iA0>(a0), Butsign<iA0>()));
```
Comment (jfalcou) [inline include/boost/simd/arch/x86/avx/simd/function/any.hpp]: those parens are unrequited, did you wanted to write a cast to bool ?

File: include/boost/simd/constant/butsign.hpp [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+  Distributed under the Boost Software License, Version 1.0.
+  (See accompanying file LICENSE.md or copy at http://boost.org/LICENSE_1_0.txt)
+*/
+//==================================================================================================
+#ifndef BOOST_SIMD_CONSTANT_BUTSIGN_HPP_INCLUDED
+#define BOOST_SIMD_CONSTANT_BUTSIGN_HPP_INCLUDED
+
+#if defined(DOXYGEN_ONLY)
+namespace boost { namespace simd
+{
```
Comment (jfalcou) [inline include/boost/simd/constant/butsign.hpp]: I think this constant should live in detail/ with the other internal ones

File: include/boost/simd/arch/x86/avx/simd/function/any.hpp (hunk 1 above)
Comment (jtlap) [reply include/boost/simd/arch/x86/avx/simd/function/any.hpp]: just copy paste remainings. Corrected

File: include/boost/simd/arch/x86/avx/simd/function/any.hpp (hunk 2 above)
Comment (jtlap) [reply include/boost/simd/arch/x86/avx/simd/function/any.hpp]: Corrected

**Comment by jfalcou** [comment]
on sse 4.1, how is the speed ?

**Comment by jtlap** [comment]
back to the preceeding faulty speed

**Comment by jfalcou** [comment]
OK. Would you mind check if clang is happy with Butsign by checking if this PR passes using clang ?

**Comment by jtlap** [comment]
ok, I look to it

**Comment by jtlap** [comment]
done and ok

**Pull request merged**